src/volar/entries/sfc-typed-router.ts

```diff
@@ -0,0 +1,76 @@
+import { relative } from 'node:path'
+import type { VueLanguagePlugin } from '@vue/language-core'
+import { replaceAll, toString } from 'muggle-string'
+import { augmentVlsCtx } from '../utils/augment-vls-ctx'
+
+const plugin: VueLanguagePlugin = (ctx) => {
+  const RE = {
+    USE_ROUTE: {
+      /** Targets the spot between `useRoute` and `()` */
+      BEFORE_PARENTHESES: /(?<=useRoute)(\s*)(?=\(\))/g,
+      /** Targets the spot right before `useRoute()` */
+      BEFORE: /(?=useRoute(\s*)\(\))/g,
+      /** Targets the spot right after `useRoute()` */
+      AFTER: /(?<=useRoute(\s*)\(\))/g,
+    },
+    DOLLAR_ROUTE: {
+      /**
+       * When using `$route` in a template, it is referred
+       * to as `__VLS_ctx.$route` in the virtual file.
+       */
+      VLS_CTX: /\b__VLS_ctx.\$route\b/g,
+    },
+  }
+
+  return {
+    version: 2.1,
+    resolveEmbeddedCode(fileName, _sfc, embeddedFile) {
+      if (!embeddedFile.id.startsWith('script_')) {
+        return
+      }
+
+      // TODO: Do we want to apply this to EVERY .vue file or only to components that the user wrote themselves?
+
+      const relativeFilePath = ctx.compilerOptions.baseUrl
```

Comment:
Will `ctx.compilerOptions.baseUrl` always equal the `VueRouter` plugin's `root`? If not, this might not always work.

Reply:
Good question